### PR TITLE
Fix mpi tests

### PR DESCRIPTION
**Pull request by gcongiu**
This PR fixes several warnings detected when compiling tests with `--enable-stricttest`. It also fixes the help message of datatype tests and removes an unused line in the `basictypetest.txt` configuration file.

### CODE REVIEW

**Comment by gcongiu** [comment]
test:jenkins/ch3/tcp

File: test/mpi/basictypetest.txt [hunk 1]
```diff
@@ -43,5 +43,4 @@ rma/lockall_dt_flushall.c::1 8 64 512 32768 262144:timeLimit=1800:4
 rma/lockall_dt_flushlocal.c::1 8 64 512 32768 262144:timeLimit=1800:4
 rma/lockall_dt_flushlocalall.c::1 8 64 512 32768 262144:timeLimit=1800:4
 rma/putfence1.c::1 8 64 512 32768 262144::2
-rma/putfence1.c:LARGE_CNT:16000000:timeLimit=1800:2
```
Comment (gcongiu) [inline test/mpi/basictypetest.txt]: @minsii this is actually incorrect, what is not used is the `LARGE_CNT` macro not the line. I will fix this if you agree with other changes.

Comment (minsii) [reply test/mpi/basictypetest.txt]: Except the comment about the warning-fix commits, the other changes look good to me.

**Comment by gcongiu** [comment]
@minsii sure we can merge them I just separated them because DTPools is separate from the test suite

**Comment by minsii** [comment]
Ah, I did not notice that they fix different portions. I think separating the commits is better.

**Pull request merged**